### PR TITLE
Adding update operation for coverages api

### PR DESCRIPTION
**Pull request by asfram**

### CODE REVIEW

File: tests/integration/rest_api_test.py [hunk 1]
```diff
@@ -98,3 +98,10 @@ def test_kown_version_status(app, monkeypatch):
     r = to_json(raw)
     assert raw.status_code == 200
     assert r.get('version') == version
+
+def test_update_coverage_returns_success_status(app):
```
Comment (prhod) [inline tests/integration/rest_api_test.py]: this function should be moved to the "coverage_api_test.py" when the PR https://github.com/CanalTP/tartare/pull/32 is merged.

File: tests/integration/rest_api_test.py [hunk 2]
```diff
@@ -98,3 +98,10 @@ def test_kown_version_status(app, monkeypatch):
...
+    raw = app.put('/coverages', headers={'Content-Type':'application/json'}, data=json.dumps({"id": "id_test", "name":"new_name_test"}))
```
Comment (prhod) [inline tests/integration/rest_api_test.py]: The test is modifying a coverage not initializes

File: tartare/core/models.py [hunk 3]
```diff
@@ -60,3 +60,8 @@ def find(cls, filter={}):
     def create_from_mongo(cls, raw):
         return Coverage(_id=raw['_id'], name=raw['name'])
 
+    @classmethod
+    def update(self, dataset):
+        result = mongo.db[self.mongo_collection].update_one({'_id': dataset['id']}, {'$set': {'name': dataset['name']}})
+        return result.raw_result
```
Comment (prhod) [inline tartare/core/models.py]: why do you return a raw_result ?

File: tests/integration/rest_api_test.py (hunk 1 above)
Comment (asfram) [inline tests/integration/rest_api_test.py]: DONE

File: tartare/core/models.py [hunk 4]
```diff
@@ -66,3 +66,12 @@ def create_from_mongo(cls, raw):
             return Coverage(_id=raw['_id'], name=raw['name'])
         else:
             return None
+
+    @classmethod
+    def update(self, dataset):
+        raw = mongo.db[self.mongo_collection].update_one({'_id': dataset['id']}, {'$set': {'name': dataset['name']}})
+        if raw.modified_count == 1:
+            return Coverage(_id=dataset['id'], name=dataset['name'])
```
Comment (antoine-de) [inline tartare/core/models.py]: couldn't you use `create_from_mongo` ?

File: tests/integration/rest_api_test.py (hunk 2 above)
Comment (asfram) [inline tests/integration/rest_api_test.py]: Test has been 'revised and edited'

File: tartare/interfaces/coverages.py [hunk 5]
```diff
@@ -66,3 +66,20 @@ def delete(self, coverage_id):
         if c == 0:
             abort(404)
         return {'coverage': None}, 204
+
+    def put(self):
+        parser = reqparse.RequestParser()
+        parser.add_argument('id', required=True, help='id is required', location='json')
+        parser.add_argument('name', required=True, help='name is required', location='json')
```
Comment (antoine-de) [inline tartare/interfaces/coverages.py]: name is not required yes ?

the id should be the only required argument, all the rest is optional and update only if provided no ?

File: tartare/core/models.py (hunk 3 above)
Comment (asfram) [inline tartare/core/models.py]: DONE: Code has been 'revised and edited'

File: tartare/core/models.py (hunk 4 above)
Comment (asfram) [inline tartare/core/models.py]: DONE: `create_from_mongo` used

File: tartare/interfaces/coverages.py (hunk 5 above)
Comment (asfram) [inline tartare/interfaces/coverages.py]: DONE: name is now and patch used instead of put

**Pull request merged**